src/toxcore/dht/server.rs

```diff
@@ -982,4 +1004,22 @@ mod tests {
         let (_, lan_discovery) = LanDiscovery::from_bytes(&buf[..size]).unwrap();
         assert_eq!(alice.pk, lan_discovery.pk);
     }
+    // remove_timedout_client()
+    #[test]
+    fn server_remove_timedout_client_test() {
```

Comment:
Add one more test that client with ping response (that was 1 sec ago) is not removed from peer cache.